c_glib/arrow-glib/memory-pool.cpp

```diff
@@ -0,0 +1,194 @@
+                                guint prop_id,
+                                GValue *value,
+                                GParamSpec *pspec)
+{
+  switch (prop_id) {
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_memory_pool_init(GArrowMemoryPool *object)
+{
+  auto priv = GARROW_MEMORY_POOL_GET_PRIVATE(object);
+  new(&priv->memory_pool) arrow::MemoryPool*;
+}
+
+static void
+garrow_memory_pool_class_init(GArrowMemoryPoolClass *klass)
+{
+  GParamSpec *spec;
+
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->set_property = garrow_memory_pool_set_property;
+  gobject_class->get_property = garrow_memory_pool_get_property;
+
+  spec = g_param_spec_pointer("memory-pool",
+                              "Memory Pool",
+                              "The raw arrow::MemoryPool* *",
+                              static_cast<GParamFlags>(G_PARAM_WRITABLE |
+                                                       G_PARAM_CONSTRUCT_ONLY));
+  g_object_class_install_property(gobject_class, PROP_MEMORY_POOL, spec);
+}
+
+/**
+ * garrow_default_memory_pool:
+ * 
+ * Returns: (transfer none): The process-wide default memory pool.
+ *
+ * Since: 9.0.0
+ */
+GArrowMemoryPool *
+garrow_default_memory_pool()
+{
+  auto memory_pool = arrow::default_memory_pool();
+  return garrow_memory_pool_new_raw(memory_pool);
+}
+
+/**
+ * garrow_memory_pool_get_bytes_allocated:
+ * @memory_pool: A #GArrowMemoryPool.
+ *
+ * Returns: The number of bytes that were allocated and not yet free’d
+ * through this allocator.
+ *
+ * Since: 9.0.0
+ */
+gint64
+garrow_memory_pool_get_bytes_allocated(GArrowMemoryPool *memory_pool)
+{
+  auto arrow_memory_pool = garrow_memory_pool_get_raw(memory_pool);
+  return arrow_memory_pool->bytes_allocated();
+}
+
+/**
+ * garrow_memory_pool_get_max_memory:
+ * @memory_pool: A #GArrowMemoryPool.
+ * 
+ * Return peak memory allocation in this memory pool.
+ *
+ * Returns: Maximum bytes allocated. If not known (or not implemented),
+ *   returns -1
+ *
+ * Since: 9.0.0
+ */
+gint64
+garrow_memory_pool_get_max_memory(GArrowMemoryPool *memory_pool)
+{
+  auto arrow_memory_pool = garrow_memory_pool_get_raw(memory_pool);
+  return arrow_memory_pool->max_memory();
+}
+
+/**
+ * garrow_memory_pool_get_backend_name:
+ * @memory_pool: A #GArrowMemoryPool.
+ * 
+ * The name of the backend used by this MemoryPool (e.g. “system” or
+ * “jemalloc”).
+ *
+ * Since: 9.0.0
+ */
+gchar *
+garrow_memory_pool_get_backend_name(GArrowMemoryPool *memory_pool)
+{
+  auto arrow_memory_pool = garrow_memory_pool_get_raw(memory_pool);
+  return g_strdup(arrow_memory_pool->backend_name().c_str());
+}
+
+G_END_DECLS
+
+GArrowMemoryPool *
+garrow_memory_pool_new_raw(arrow::MemoryPool *memory_pool)
+{
+  return garrow_memory_pool_new_raw_bytes(memory_pool);
+}
+
+GArrowMemoryPool *
+garrow_memory_pool_new_raw_bytes(arrow::MemoryPool *memory_pool)
```

Comment:
We don't need `garrow_memory_pool_new_raw_bytes()`.

```suggestion
GArrowMemoryPool *
garrow_memory_pool_new_raw(arrow::MemoryPool *memory_pool)
```